libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -535,6 +535,21 @@ void AP_Baro::init(void)
                                       std::move(hal.spi->get_device(HAL_BARO_LPS22H_NAME))));
 #endif
 
+#ifdef HAL_BARO_MS5611_I2C_BUS
```

Comment:
Why are all these lines below needed instead of each board simply defining HAL_BARO_DEFAULT? Unless I missed something, this code is repeated above, except it checks HAL_BARO_DEFAULT value.

Reply:
because most boards don't have a baro so user solders it itself, so we don't know what baro will be